RichString.c

```diff
@@ -54,9 +54,45 @@ void RichString_rewind(RichString* this, int count) {
 
 #ifdef HAVE_LIBNCURSESW
 
+static size_t mbstowcs_nonfatal(wchar_t* dest, const char* src, size_t n) {
+   wchar_t *d = dest;
```

Comment:
1. Style
```suggestion
   wchar_t* d = dest;
```

2. I didn't get why you are duplicating two pointers `src` and `dest` with `s` and `d` respectively where you can just increment `src` and `dest` directly.

Reply:
Same goes for `remaining` vs `n`.

`s` and `d` are not very descriptive.

Also arguments are by-value, thus you can change the pointers locally without issues.